airflow/providers_manager.py

```diff
@@ -46,42 +45,134 @@ class ProvidersManager:
     """Manages all provider packages."""
 
     def __init__(self):
+        self.__log = logging.getLogger(__name__)
```

Comment:
(Not important, but why so private?)

Reply:
Privacy by design. I switched to the (recommended by many) mode of programming - only expose what you expect to be used and everything else private by design. That makes it much more encapsulated and makes it easier to avoid accidental uses and unintended side effects.